cpp/include/cuspatial/polygon_bounding_box.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/types.hpp>
+
+#include <memory>
+
+namespace cuspatial {
+
+/**
+ * @brief Compute minimum bounding boxes for a set of polygons.
+ *
+ * @param poly_offsets Begin indices of the first ring in each polygon (i.e. prefix-sum)
+ * @param ring_offsets Begin indices of the first point in each ring (i.e. prefix-sum)
+ * @param x Polygon point x-coordinates
+ * @param y Polygon point y-coordinates
+ *
+ * @return a cudf table of bounding boxes as four columns of the same type as `x` and `y`:
+ * x1 - the lower-left x-coordinate of each bounding box
+ * y1 - the lower-left y-coordinate of each bounding box
+ * x2 - the upper-right x-coordinate of each bounding box
+ * y2 - the upper-right y-coordinate of each bounding box
```

Comment:
But this is what we call it in all the other methods that return bboxes?

Reply:
I'm strongly in favor of using min/max as opposed to left/right/top/bottom, because the dictated orientation may not be the same for all GIS we work with.